src/Widgets/DiskBar.vala

```diff
@@ -24,113 +24,88 @@ public class Installer.DiskBar: Gtk.Grid {
-        label.orientation = Gtk.Orientation.VERTICAL;
-        label.row_spacing = 6;
-        label.margin_end = 12;
-        label.valign = Gtk.Align.CENTER;
-        label.add (name_label);
-        label.add (size_label);
-
-        var bar = new Gtk.Grid ();
-        bar.width_request = 40;
-
-        var bar_style_context = bar.get_style_context ();
-        bar_style_context.add_class (Gtk.STYLE_CLASS_TROUGH);
-        bar_style_context.add_class ("disk-bar");
-
-        bar.size_allocate.connect ((alloc) => {
-            update_sector_lengths (partitions, alloc);
-        });
+        show_all ();
+    }
 
-        foreach (PartitionBar part in partitions) {
-            bar.add (part);
+    private uint64 get_unused () {
+        uint64 used = 0;
+        foreach (PartitionBar partition in partitions) {
+            used += partition.get_size ();
         }
 
-        legend_container = new Gtk.Grid ();
-        legend_container.column_spacing = 12;
-        legend_container.halign = Gtk.Align.CENTER;
-        legend_container.margin_bottom = 9;
+        return size - (used * 512);
+    }
 
-        var legend = new Gtk.ScrolledWindow (null, null);
+    private void generate_legend () {
+        legend = new Gtk.ScrolledWindow (null, null);
         legend.vscrollbar_policy = Gtk.PolicyType.NEVER;
+
+        legend_container = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 12);
+        legend_container.set_halign (Gtk.Align.CENTER);
+        legend_container.margin_bottom = 9;
         legend.add (legend_container);
 
         foreach (PartitionBar p in partitions) {
-            add_legend (p.path, p.get_size () * 512, Distinst.strfilesys (p.filesystem), p.vg, p.menu);
-        }
-
-        uint64 used = 0;
-        foreach (PartitionBar partition in partitions) {
-            used += partition.get_size ();
+            add_legend (p.path, p.get_size() * 512, Distinst.strfilesys (p.filesystem), p.vg, p.menu);
         }
 
-        var unused = size - (used * 512);
         if (size / 100 < unused) {
             add_legend ("unused", unused, "unused", null, null);
-
-            var unused_bar = new Gtk.Grid ();
-            unused_bar.expand = true;
-
-            var unused_style_context = unused_bar.get_style_context ();
-            unused_style_context.add_class ("fill-block");
-            unused_style_context.add_class ("unused");
-
-            bar.add (unused_bar);
         }
-
-        hexpand = true;
-        margin = 6;
-        get_style_context ().add_class (Granite.STYLE_CLASS_STORAGEBAR);
-        attach (label, 0, 1);
-        attach (legend, 1, 0);
-        attach (bar, 1, 1);
-
-        show_all ();
     }
 
     private void add_legend (string ppath, uint64 size, string fs, string? vg, Gtk.Popover? menu) {
         var fill_round = new FillRound ();
         fill_round.width_request = fill_round.height_request = 14;
-        fill_round.valign = Gtk.Align.CENTER;
+        fill_round.set_valign(Gtk.Align.CENTER);
```

Comment:
There are a lot of switching back to setter methods instead of properties. This is probably just a weird merge artifact.